### DISCUSSION TITLE
Build & Deployment error

### DISCUSSION THREAD

**Discussion by developerdesigner18** [vercel/community#4289] [Help]
### Summary

I am getting "Error: error:0308010C:digital envelope routines::unsupported" error while creating build of my reactjs application. Things I tried :

Upgrading build script to :  "set NODE_OPTIONS=--openssl-legacy-provider && react-scripts build" which also gives while creating production build

Upgrading react-scripts version to 5+ 

### Steps to Reproduce

https://prnt.sc/ZT6cGePzaY0s

**Comment by amyegan** [comment]
Please try removing "set" from the build command and make it `NODE_OPTIONS=--openssl-legacy-provider && react-scripts build`. Alternatively, you can remove `NODE_OPTIONS` from your build command and make it an [environment variable](https://vercel.com/docs/projects/environment-variables) with the value `--openssl-legacy-provider`. 🙏

**Reply by dezzyboy** [reply]
just thought i would share this oddly, using the environment variable on vercel env section does not work for some reason vercel is not reading it. to fix this adding `node-options="--openssl-legacy-provider"` to a .npmrc file at the root of your project ensures that npm consistently recognizes and applies it when executing scripts. hopes this helps anyone else in the future cheers
